File: src/main/java/net/ndrei/teslacorelib/items/BaseAddonItem.java
package net.ndrei.teslacorelib.items;

import net.minecraft.item.ItemStack;
import net.minecraft.item.crafting.IRecipe;
import net.minecraftforge.oredict.ShapedOreRecipe;
import net.ndrei.teslacorelib.TeslaCoreLib;

/**
 * Created by CF on 2017-01-02.
 */
public class BaseAddonItem extends BaseAddon {
    public BaseAddonItem() {
        super(TeslaCoreLib.MODID, TeslaCoreLib.creativeTab, "base_addon");
    }

    @Override
    protected IRecipe getRecipe() {
        return new ShapedOreRecipe(new ItemStack(this, 1),
                "xyx", "xxx", "xyx",
                'x', "paper", // Items.PAPER,
                'y', "dustRedstone" // Items.REDSTONE
        );
    }
}

File: src/main/kotlin/net/ndrei/teslacorelib/items/BaseAddonItem.kt
package net.ndrei.teslacorelib.items

import net.ndrei.teslacorelib.ProxyLoadLevel
import net.ndrei.teslacorelib.TeslaCoreLib
import net.ndrei.teslacorelib.annotations.AutoRegisterItem
import net.ndrei.teslacorelib.config.TeslaCoreLibConfig

/**
 * Created by CF on 2017-06-27.
 */
@AutoRegisterItem(TeslaCoreLibConfig.REGISTER_ADDONS)
object BaseAddonItem
    : BaseAddon(TeslaCoreLib.MODID, TeslaCoreLib.creativeTab, "base_addon") {
    init {
        TeslaCoreLib.proxy.testLoadLevel(ProxyLoadLevel.ITEMS, this.javaClass.simpleName)
    }
}
